parsers/src/main/scala/uk/gov/ons/addressIndex/parsers/Tokens.scala

```diff
@@ -1,29 +1,67 @@
 package uk.gov.ons.addressIndex.parsers
 
-import uk.gov.ons.addressIndex.crfscala.CrfScala.{CrfToken, CrfTokenable, CrfTokens, Input}
+import com.typesafe.config.ConfigFactory
+import uk.gov.ons.addressIndex.crfscala.CrfScala.CrfTokenable
 
-//todo scaladoc this thing
+import scala.io.Source
+
+/**
+  * Tokenizer for the input
+  */
 object Tokens extends CrfTokenable {
 
-  type TokenIndicator = String
-  type Token = CrfToken
-  type Tokens = CrfTokens
+  private val config = ConfigFactory.load()
+
+  /**
+    * Tokenizes input into tokens (also removes counties, replaces synonyms)
+    * @param input the string to be tokenized
+    * @return Array of tokens
+    */
+  override def apply(input: String): Array[String] ={
+    val upperInput = input.toUpperCase()
+
+    val inputWithoutCounties = removeCounties(upperInput)
+
+    val tokens = inputWithoutCounties
+      .replace(" - ", "-")
+      .replace("- ", "-")
+      .replace(" -", "-")
```

Comment:
I think this is what you might want to use - its catches space from the hyphen to user input
`\d+\s?.*?-.*?\s?\d+`